src/py42/_internal/clients/securitydata.py

```diff
@@ -6,3 +6,66 @@ def get_security_event_locations(self, user_uid):
         uri = u"/c42api/v3/SecurityEventsLocation"
         params = {u"userUid": user_uid}
         return self._session.get(uri, params=params)
+
+
+class SavedSearchClient(BaseClient):
+    """A client to interact with saved search APIs."""
+
+    _version = u"v1"
+    _resource = u"/forensic-search/queryservice/api/{}/saved".format(_version)
+
+    _response_header = [u"name", u"id", u"notes", u"modifiedByUsername", u"modifiedTimestamp"]
+
+    def __init__(self, session, file_event_client):
+        super(SavedSearchClient, self).__init__(session)
+        self._file_event_client = file_event_client
+
+    def _parse_response(self, response):
+        if u"searches" in response:
+            return {
+                key: search[key]
+                for search in response[u"searches"]
+                for key in self._response_header
+            }
+
+    def get(self):
+        """Fetch details of existing saved searches.
+        Provides details such as saved search id, name, notes, modifier name, last modified time.
+
+        Returns:
+            dict
+        """
+        uri = u"{}".format(self._resource)
+        response = self._session.get(uri)
+        return self._parse_response(response)
+
+    def get_by_id(self, search_id):
+        """Fetch details of saved search for given search Id.
+
+        Args:
+            search_id (str): Unique search Id of the saved search, Required.
+        Returns:
+            :class:`py42.response.Py42Response`
+        """
+        uri = u"{}/{}".format(self._resource, search_id)
+        return self._session.get(uri)
+
+    @staticmethod
+    def _make_query(response):
```

Comment:
Right now `execute` method fails with `BadRequest` error,  I have tried various ways to make the desired query but still can't get it right. 

`return FileEventQuery(*response[u"searches"][0][u"groups"])`.  Need to fix this, still working on it.

Reply:
I can help with this if need be

Reply:
Yes, Please, I have already spent few hours on it, comparing the request against web-console.

Reply:
When I run with debug mode, I get this data "{'pgNum': 1, 'pgSize': 100, 'purpose': 'USER_EXECUTED_SEARCH', 'groups': [{'filters': [{'term': 'eventTimestamp', 'operator': 'ON_OR_AFTER', 'value': '2020-05-01T00:00:00.000Z', 'display': None}], 'filterClause': 'AND', 'display': '{\"data\":{\"isMultivalue\":false},\"version\":\"v1\"}'}, {'filters': [{'term': 'eventType', 'operator': 'IS', 'value': 'DELETED', 'display': None}, {'term': 'eventType', 'operator': 'IS', 'value': 'EMAILED', 'display': None}, {'term': 'eventType', 'operator': 'IS', 'value': 'MODIFIED', 'display': None}, {'term': 'eventType', 'operator': 'IS', 'value': 'READ_BY_APP', 'display': None}, {'term': 'eventType', 'operator': 'IS', 'value': 'CREATED', 'display': None}], 'filterClause': 'OR', 'display': '{\"data\":{\"isMultivalue\":true},\"version\":\"v1\"}'}]}"

The `display: None` seems concerning

Reply:
I think we can just remove `display`. Isn't that a UI property? Also, https://py42docs.code42.com/en/stable/userguides/gettingstarted.html#troubleshooting-and-support shows how to turn on debug mode. It is really useful when you are not getting the expected status codes because it prints request parameters to stdout

Reply:
FYI I was able to get this working, ended up adding some other useful things. Adding some tests and such and I'll make it available shortly.

Reply:
Actually, if you convert these to be "null", instead of null, I think it should work..